app/models/reference_repository.rb

```diff
@@ -0,0 +1,65 @@
+# frozen_string_literal: true
+
+class ReferenceRepository < ApplicationRecord
```

Comment:
You may wish to bring in the Indexable Concern.
It does bring a lot with it so I appreciate it's fairly complex, but it will give you the stuff regarding switching back and front indexing (if desired)
And also has some helper functions for searching / aggregation though granted it is a bit mixed up between the various classes (something that i'd like to refactor one day)

Reply:
I brought it in.
Works well